src/main/kotlin/auth/BasicAuth.kt

```diff
@@ -0,0 +1,78 @@
+package se.zensum.leia.auth
+
+import com.mantono.pyttipanna.HashAlgorithm
+import com.mantono.pyttipanna.hash
+import io.ktor.util.decodeBase64
+import leia.logic.IncomingRequest
+import mu.KotlinLogging
+import org.apache.commons.codec.binary.Hex
+import java.util.*
+
+private val log = KotlinLogging.logger("basic-auth")
+
+private const val HEADER = "Authorization"
+
+class BasicAuth(credentials: Map<String, String>): AuthProvider {
+
+    private val credentials: Map<String, ByteArray> = credentials.mapValues {
+        Hex.decodeHex(it.value)
+    }
+
+    init {
+        this.credentials.values.forEach { hash ->
+            require(hash.size == 32) {
```

Comment:
Wrong password in auth header will not yield a crash, but loading a config where the hash is not a valid length for a sha256 will lead to a crash, yes. Is this not preferable? I cannot see that we would ditch the authorization for a routes if the hashes are invalid (that does not seem safe), maybe deny all access to the route maybe instead of crashing?

Reply:
Okey.